### ISSUE TITLE
"TypeError: Cannot perform 'get' on a proxy that has been revoked thrown" in Object.assign

### ISSUE THREAD

**Issue by HiroP0** [gullerya/object-observer#8]
__Problem__: TypeError: Cannot perform 'get' on a proxy that has been revoked thrown in Object.assign.

minimal code:
```javascript
var oldData = { b : {b1 : "x", b2 : "y"  } };
var newData = { b : { b1: "z" } };

var observableData = require("object-observer").Observable.from(oldData);
observableData.observe(dataObserver);
Object.assign(observableData, newData);

function dataObserver(changes) {
  changes.forEach(change => {
    console.log(change.type + " of " + change.path + " from " + change.value + " to " + change.oldValue);
  });
};
```
__Proposal__:
I'm javascript newbie, so no proposals. Error might be on my side.

Relevant __version/s__ (if any): 0.2.4

(node v8.9.4)

**Comment by gullerya** [comment]
I think I see the problem, but would like to ensure that we are on the same page :).

If I'm running your case, which in itself an interesting and absolutely correct one, while the `observer` function looks like:
```
function dataObserver(changes) {
   changes.forEach(change => console.dir(change));
}
```

i'm not getting any exception, nor in browser neither in Node, but the `oldValue` that is being returned as part of the `change` object is actually empty.

If you'll be able to spend few more minutes and confirm that, than I'll adjust the issue to more precise one and will better explain what is happening.

Thanks in advance.

**Comment by HiroP0** [comment]
With your observer function I'm not getting the error either. Message appears as

```
UpdateChange {}
```
indicating an empty object I suppose. Debugger shows that a change object with correct oldValue is going in.

Looking at the ```UpdateChange``` object in my observer function in a debugger I too see its ```oldValue```  as 

```
Proxy { b1: "x", b2 : "y} 
``` 

https://stackoverflow.com/questions/11954152/whats-the-difference-between-console-dir-and-console-log is interesting reading. All new to me.

Cheers,
Michael

**Comment by gullerya** [comment]
Fixed in 0.2.5
The following test case is part of the tests suite and covering this issue:
```
suite.addTest({name: 'Object.assign on observable should raise an event/s of update with non-proxified original object'}, function(pass, fail) {
	let oldData = {b: {b1: "x", b2: "y"}},
		newData = {b: {b1: "z"}},
		observable = Observable.from(oldData),
		events = [];
	observable.observe(changes => events.push.apply(events, changes));
	Object.assign(observable, newData);
	if (events.length !== 1 || events[0].type !== 'update' || events[0].path.join('.') !== 'b') {
		fail('UPDATE event metadata NOT as expected');
	} else if (!events[0].value || events[0].value.b1 !== 'z') {
		fail('UPDATE event (new) value NOT as expected');
	} else if (!events[0].oldValue || events[0].oldValue.b1 !== 'x' || events[0].oldValue.b2 !== 'y') {
		fail('UPDATE event oldValue NOT as expected');
	}

	pass();
});
```

Thanks goes to @HiroP0 for the time and the efforts.
I'm closing this one, please feel free to re-open/feed-back if this one is still not working for you.

**Comment by HiroP0** [comment]
Works!
Quick and thorough.
Cheers.
